Assets/MRTK/Core/Providers/Hands/BaseHandVisualizer.cs

```diff
@@ -192,12 +192,16 @@ public virtual void OnHandMeshUpdated(InputEventData<HandMeshInfo> eventData)
 
                 bool newMesh = handMeshFilter == null;
 
+                IMixedRealityInputSystem inputSystem = CoreServices.InputSystem;
+                MixedRealityHandTrackingProfile handTrackingProfile = inputSystem?.InputSystemProfile != null ? inputSystem.InputSystemProfile.HandTrackingProfile : null;
                 if (newMesh &&
-                    CoreServices.InputSystem?.InputSystemProfile != null &&
-                    CoreServices.InputSystem.InputSystemProfile.HandTrackingProfile != null &&
-                    CoreServices.InputSystem.InputSystemProfile.HandTrackingProfile.HandMeshPrefab != null)
+                    handTrackingProfile != null &&
+                    handTrackingProfile.SystemHandMeshMaterial != null)
```

Comment:
Since it looks like we aren't currently removing the old handMeshPrefab from the profile serialization, do we want to fall back to it if the dev _hasn't_ set SystemHandMeshMaterial but still has the old field set? Could be a decent back compat scenario

Reply:
Fixed!

Reply:
Done